lib/board.rb

```diff
@@ -1,42 +1,64 @@
 class Board
 
-  def create_grid(dimension)
-    size = Math.sqrt(dimension)
+  attr_reader :dimension, :grid
+
+  def initialize(dimension)
+    @dimension = dimension
+    @grid = create_grid
+  end
+
+  def create_grid
+    possible_rows.inject({}) do |result, element|
+      result[element] = possible_columns
+      result
+    end
+  end
+
+  def size_of_board
+    Math.sqrt(@dimension)
+  end
+
+  def possible_rows
     alphabet = ("A".."Z").to_a
-    letters = alphabet[0..size - 1]
-    letters.inject({}) {|result, element| result[element] = (1..size).to_a; result }
+    alphabet[0..size_of_board - 1]
+  end
+
+  def possible_columns
+    (1..size_of_board).to_a
   end
 
-  def mark_board(grid, row, column, mark)
-    grid[row][column - 1] = mark
-    grid
+  def grid_with_one_hardcoded_ship
+    mark_board("C", 2, ship_mark)
+    mark_board("C", 3, ship_mark)
+    @grid
   end
 
-  def place_ship(grid)
-    mark_board(grid, "C", 2, ship_mark)
-    mark_board(grid, "C", 3, ship_mark)
-    grid
+  def mark_board(row, column, mark)
+    @grid[row][column - 1] = mark
+    @grid
   end
 
   def ship_mark
-    "-"
+    Peg.new.ship
   end
 
-  def hit_or_miss(grid_with_ships, row, column)
+  def hit?(grid_with_ships, row, column)
     coordinate = grid_with_ships[row][column - 1]
-    if coordinate == ship_mark
-      "hit"
-    else
-      "miss"
-    end
+    coordinate == ship_mark
   end
 
-  def hit_all(grid_with_ships)
+  def hit_all?(grid_with_ships)
     marks = grid_with_ships.values.flatten
-    if marks.include?(ship_mark)
-      "not hit all"
-    else
-      "hit all"
+    !marks.include?(ship_mark)
+  end
+
+  def mark_board_with_ship(coordinates)
+    length_of_boat = coordinates.count - 1
+    index = 1
+    until index > length_of_boat
```

Comment:
I'm not sure what the shape of `coordinates` is, but can you explore using a Ruby Enumerable here instead of the while loop?